### DISCUSSION TITLE
HmIP-FWI Wiegand Interface - detected gaps / issues

### DISCUSSION THREAD

**Discussion by Bastian007** [SukramJ/aiohomematic#1285] [General]
Hi Daniel,
for my HmIP-FWI (Wiegand Interface), I observerd two issues with hahomematic integration in Home assistant:

1. Numeric Code ID sensor shows strange values, such as 31.0

![image](https://github.com/danielperna84/hahomematic/assets/12425311/c341f82a-66ae-42d0-bd75-4978d3696fe1)
<img width="379" alt="image" src="https://github.com/danielperna84/hahomematic/assets/12425311/8679130b-312d-4ad1-b5f8-9a322b82bab6"><img width="437" alt="image" src="https://github.com/danielperna84/hahomematic/assets/12425311/7d30bb5c-0a31-4ecc-9992-e66040199a35">

Potentially this is not an issue of the integration to HA as I've found this:
https://forum.iobroker.net/topic/55966/hmip-fwi-code_id-has-value-31-greater-than-max-20
And the log in CCU3 shows this
<img width="788" alt="image" src="https://github.com/danielperna84/hahomematic/assets/12425311/c00458d2-d1b7-4ad3-951b-b8792a24f400">

But according to the min/max values it shoul become only an integer between 1 and 21.
So probably I am going to open a request to eQ-3 to get clarity on those codes.

2. Channel 0 is not provided

<img width="628" alt="image" src="https://github.com/danielperna84/hahomematic/assets/12425311/d7fb236c-8fbd-4a13-b025-82d5a447fa09">
<img width="255" alt="image" src="https://github.com/danielperna84/hahomematic/assets/12425311/07b73854-a2be-4b3a-9bfd-6ed7f488891c"><img width="250" alt="image" src="https://github.com/danielperna84/hahomematic/assets/12425311/55a64395-e83b-4e4e-b57e-67a3109b81c3"><img width="257" alt="image" src="https://github.com/danielperna84/hahomematic/assets/12425311/81fdf671-6b76-42cc-8462-ef73877e07ed">

Especially the door ring status would be great to have in HA available.
Workaround currently is to set a system variable according to those status at CCU3 which is then reflected in HA as well (with some time delay).

Would be great if you could have a look to those items. I am also happy to help to sort this out.
Thanks & best regards, Bastian

**Comment by SukramJ** [comment]
1. We usually don't care about sensor values. Sensor values are displayed as delivered. Only when writing, which does not happen with the sensor, is the value checked against min/max.
2.  We don't have a device definition for the HmIP-FWI, so you can please [create](https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition) one and attach it here or better create a PR at pydevccu.
3. Since I know the device definition I can only guess, but I would guess that door ring status is available as an event or device trigger.

**Reply by SukramJ** [reply]
The [paramset description](https://github.com/danielperna84/pydevccu/issues/46#:~:text=HmIP%2DFWI_export_paramset_descriptions.json), i.e. the values under the VALUES keys, contain no reference to anything like a doorbell. This seems to be defined in the CCU UI, but not in the XmLRPC API. You can take a look at it yourself.
The device in HA looks like this, but all displayable entities are also displayed. Maybe one of the sensors reacts to a doorbell being pressed.

**Reply by Bastian007** [reply]
For others searching for the same:
When input for connecting a doorbell button is leveraged at FWI, the Entity Id number.hmip_fwi_code_id contains value 21 when the doorbell is pressed.

probably as well a helpful reference: https://homematic-forum.de/forum/viewtopic.php?p=786310#p786500
